qubes/tests/app.py

```diff
@@ -1096,6 +1116,73 @@ def test_207_default_kernel(self):
                 existence.side_effect = [True, False]
                 self.app.default_kernel = "unittest_GNU_Hurd_1.0.0"
 
+    def test_300_preload_default_dispvm(self):
+        """Fire event for new setting from no previous one."""
+        self.appvm.features["preload-dispvm-max"] = "1"
+        with mock.patch.object(self.appvm, "fire_event_async") as mock_events:
+            self.app.default_dispvm = self.appvm
+            mock_events.assert_called_once()
+            self.assertEqual(
+                "domain-preload-dispvm-start", mock_events.call_args[0][0]
+            )
+
+    def test_301_preload_default_dispvm_change(self):
+        """Fire event for old and new setting."""
+        self.appvm.features["preload-dispvm-max"] = "1"
+        self.app.default_dispvm = self.appvm
+        self.appvm_alt.features["preload-dispvm-max"] = "1"
+        with mock.patch.object(
+            self.appvm, "fire_event_async"
+        ) as mock_old, mock.patch.object(
+            self.appvm_alt, "fire_event_async"
+        ) as mock_new:
+            self.app.default_dispvm = self.appvm_alt
+            mock_old.assert_called_once()
+            mock_new.assert_called_once()
+            self.assertEqual(
+                "domain-preload-dispvm-start", mock_new.call_args[0][0]
+            )
```

Comment:
```suggestion
            mock_new.assert_called_once_with("domain-preload-dispvm-start")
```
If  there are other args that are not interesting for test, you can use `unittest.mock.ANY` as a placeholder.
Besides being easier to read, it's also easier to spot mistakes like too much copy&paste - see the one below checks `mock_new` again...

Reply:
Implemented with `mock.ANY` locally, much better.